lib/services/devices/deviceService.js

```diff
@@ -337,36 +338,44 @@ function updateRegisterDevice(deviceObj, callback) {
         }
     }
 
-    function extractDeviceDifference(newDevice, oldDevice, callback) {
-        var deviceData = {
-                id: oldDevice.id,
-                name: oldDevice.name,
-                type: oldDevice.type,
-                service: oldDevice.service,
-                subservice: oldDevice.subservice,
-                commands: [],
-                staticAttributes: []
-            },
-            oldActiveKeys,
+    function mixAttributeArrays(oldArray, newArray) {
```

Comment:
I am not sure if this is the best name for this function taking into consideration what it does... :) Shouldn't it be something such as `getNewAttributes()` since it returns attributes included in `newArray` but not included in `oldArray` (if I am not wrong :))